From 73254df370fec6a2eb1ae6711955307a1d4bf173 Mon Sep 17 00:00:00 2001
From: Ramprasad N <x0038811@ti.com>
Date: Fri, 13 Oct 2017 15:48:06 +0530
Subject: [PATCH] v4l2src: Increase minimum num buffers by 4

v4l2src plugin allocates 4 buffers for the pipeline.
To capture video in realtime the pipeline without frameloss,
four buffers are not sufficient. Increase the
number of buffers by 4 more.

Signed-off-by: Ramprasad N <x0038811@ti.com>
---
 sys/v4l2/gstv4l2object.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 549e06f..773f46d 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -3939,7 +3939,9 @@ gst_v4l2_object_decide_allocation (GstV4l2Object * obj, GstQuery * query)
      * to fill the pipeline, the minimum required to decoder according to the
      * driver and 1 more, so we don't endup up with everything downstream or
      * held by the decoder. */
-    own_min = min + obj->min_buffers + 1;
+    /* Increase the number of buffers required by 4 more to capture video in
+     * realtime. With default number of 4 buffers, frame loss is observed */
+    own_min = min + obj->min_buffers + 5;
 
     /* If no allocation parameters where provided, allow for a little more
      * buffers and enable copy threshold */
-- 
1.9.1

